src/mapkey.cpp

```diff
@@ -18,37 +18,34 @@ size_t MapKey::size() const
 
 std::string MapKey::argument_type_list() const
 {
-  size_t n = args_.size();
-  if (n == 0)
-    return "[]";
-
   std::ostringstream list;
   list << "[";
-  for (size_t i = 0; i < n-1; i++)
-    list << args_.at(i) << ", ";
-  list << args_.at(n-1) << "]";
+  for (size_t i = 0; i < args_.size(); i++)
+  {
+    if (i)
+      list << ", ";
+    list << args_[i];
+  }
+  list << "]";
   return list.str();
 }
 
 std::string MapKey::argument_value_list(BPFtrace &bpftrace,
     const std::vector<uint8_t> &data) const
 {
-  size_t n = args_.size();
-  if (n == 0)
-    return "";
-
-  std::ostringstream list;
-  list << "[";
+  std::string list = "[";
```

Comment:
This changes the behaviour when a map has no arguments, e.g. `@ = count()`
now prints:
`@[]: 123` instead of `@: 123`

Maybe just add the `n==0` check back in?

Reply:
Ah overlooked that. Added back the empty case.